### ISSUE TITLE
CLI issue for paths with spaces in them

### ISSUE THREAD

**Issue by logic** [hashicorp/vault#783]
Small reproduction recipe attached for a CLI bug I bumped into today: [testcase.txt](https://github.com/hashicorp/vault/files/36322/testcase.txt) (please ignore the Fedora-specific fix, unless you're using golang as shipped on Fedora or RHEL. :wink:)

Backstory: I use SpiderOak for backing up my development workspace. So, git checkouts actually live in a directory named `$HOME/SpiderOak Hive` (with a space in the name). But that's a pain to type all the time, so while the real directory is `$HOME/SpiderOak Hive/devel`, I have a `$HOME/devel` symlink pointing it. GOPATH is set to `$HOME/devel/go`.

Everything builds as expected, but when you try to run the vault development server, you get:

```
$ ./vault server -dev
Error initializing dev mode: Error: exit status 126

/bin/bash: /tmp/go: Is a directory
```

And if you look under the hood:

```
$ strace -ff -s 1024 /vault server -dev
[...]
[pid 25131] execve("/bin/bash", ["/bin/bash", "-c", "/tmp/go test/src/github.com/hashicorp/vault/bin/vault token-disk store"], [/* 66 vars */] <unfinished ...>
[...]
[pid 25131] execve("/tmp/go", ["/tmp/go", "test/src/github.com/hashicorp/vault/bin/vault", "token-disk", "store"], [/* 65 vars */]) = -1 EACCES (Permission denied)
[...]
```

(CWD, at this point, is `/tmp/go/src/github.com/hashicorp/vault`, but someone somewhere along the line is doing a `readlink()` to get the real path to the vault executable.)

Whoops. :smiley: I'm still poking to see whether this is something vault is doing, or if it's behavior from one of the dependencies, but thought I'd post this much in case someone immediately recognizes what's going on.

**Comment by jefferai** [comment]
I can reproduce this, and I'm looking into it.

**Comment by jefferai** [comment]
I'll try to get a fix in for 0.4 but can't be sure...in the meantime this can be worked around by running the executable from a path that doesn't contain a space, since it's logic building the token helper that isn't taking spaces into consideration, and that depends on the runtime path.

**Comment by jefferai** [comment]
I'm closing this in favor of #850, as that fixes another issue when launching dev mode that when fixed will also fix this issue. This should make it into the next release.
